pkg/kopialib/storage/s3.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	kopiaerrors "github.com/kanisterio/kanister/pkg/kopia/errors"
+	kopialibutils "github.com/kanisterio/kanister/pkg/kopialib/utils"
+	"github.com/kanisterio/kanister/pkg/utils"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/s3"
+	"github.com/pkg/errors"
+)
+
+var _ Storage = &s3Storage{}
+
+var requiredS3Arguments = []string{
+	kopialibutils.BucketKey,
+}
+
+type s3Storage struct {
+	options *s3.Options
+	create  bool
```

Comment:
should this be named `createBucketIfNotExists` to make it more readable?

Reply:
The create flag is not even used in the upstream kopia - https://github.com/kopia/kopia/blob/master/repo/blob/s3/s3_storage.go#L329. I am removing the create attribute from struct for now. Sorry for the confusion